p2p/net/swarm/addr/filter.go

```diff
@@ -0,0 +1,27 @@
+package addrutil
+
+import (
+	ma "github.com/jbenet/go-multiaddr"
+	mafmt "github.com/whyrusleeping/mafmt"
+)
+
+func SubtractFilter(addrs ...ma.Multiaddr) func(ma.Multiaddr) bool {
```

Comment:
A short godoc here would be helpful, to explain what this does and why it's useful.